decidim-core/app/controllers/concerns/decidim/needs_tos_accepted.rb

```diff
@@ -37,8 +37,7 @@ def permitted_paths?
                          decidim.delete_account_path,
                          decidim.accept_tos_path,
                          decidim.download_your_data_path,
-                         decidim.export_download_your_data_path,
-                         decidim.download_file_download_your_data_path]
```

Comment:
This removal is actually preventing a current feature related to the not acceptance of the ToS. I think there's a missing spec for detecting this (that we should add). 

Basically wit the current approach, a participant that didn't accepted a new ToS will not be able to download their data.

Steps to reproduce: 

1. Go to http://localhost:3000/admin/static_pages/terms-of-service/edit
2. Check "There have been noticeable changes." and save
3. Go to  http://localhost:3000/ - you should be redirected to http://localhost:3000/pages/terms-of-service 
4. Click on "Refuse" 
5. Click on "[download your data](http://localhost:3000/download_your_data)". Note that there's missing styling in this link, it can be fixed by adding this patch: 
```patch
diff --git a/decidim-core/app/packs/stylesheets/decidim/_modal_tos_refuse.scss b/decidim-core/app/packs/stylesheets/decidim/_modal_tos_refuse.scss
index 6d260979b9..6902ed311e 100644
--- a/decidim-core/app/packs/stylesheets/decidim/_modal_tos_refuse.scss
+++ b/decidim-core/app/packs/stylesheets/decidim/_modal_tos_refuse.scss
@@ -3,4 +3,8 @@
   [data-dialog-actions] {
     @apply flex-nowrap;
   }
+
+  a {
+    @apply underline text-secondary;
+  }
 }
```
6. Try to download your data, see that you're redirected to the ToS page again 

Adding the URL `decidim.download_download_your_data_path` for the permitted paths does not work, as it gives an error that the UUID is missing. 

If you want we can handle this issue in another PR, as I don't have a quick solution at the moment.

Reply:
an old `return true if request.path.starts_with?(decidim.download_your_data_path)` seems to do the trick.

Reply:
fixed in [8ed1f40](https://github.com/decidim/decidim/pull/13571/commits/8ed1f4002d6c6ff491f429e0dba3d7a4023614fb)